hack/lib/catalogsource.bash

```diff
@@ -12,133 +12,59 @@ function ensure_catalogsource_installed {
-    oc -n "${OLM_NAMESPACE}" create configmap serverless-bundle-sha1sums \
-      --from-file="${rootdir}/_output/serverless-bundle.sha1sum"
-    rm -f "${rootdir}/_output/serverless-bundle.sha1sum"
-  else
-    logger.info 'Serverless bundle build is up-to-date.'
-  fi
+    cp "$csv" "${rootdir}/_output/bkp.yaml"
+
+    if [ -n "$OPENSHIFT_CI" ]; then
+      # Image variables supplied by ci-operator.
+      sed -i "s,image: .*openshift-serverless-.*:knative-operator,image: ${KNATIVE_OPERATOR}," "$csv"
+      sed -i "s,image: .*openshift-serverless-.*:knative-openshift-ingress,image: ${KNATIVE_OPENSHIFT_INGRESS}," "$csv"
+      sed -i "s,image: .*openshift-serverless-.*:openshift-knative-operator,image: ${OPENSHIFT_KNATIVE_OPERATOR}," "$csv"
+    elif [ -n "$DOCKER_REPO_OVERRIDE" ]; then
+      sed -i "s,image: .*openshift-serverless-.*:knative-operator,image: ${DOCKER_REPO_OVERRIDE}/knative-operator," "$csv"
+      sed -i "s,image: .*openshift-serverless-.*:knative-openshift-ingress,image: ${DOCKER_REPO_OVERRIDE}/knative-openshift-ingress," "$csv"
+      sed -i "s,image: .*openshift-serverless-.*:openshift-knative-operator,image: ${DOCKER_REPO_OVERRIDE}/openshift-knative-operator," "$csv"
+    fi
 
-  logger.debug 'Undo potential changes to the CSV to not pollute the repository.'
-  mv "${rootdir}/_output/bkp.yaml" "$csv"
+    cat "$csv"
 
-  logger.debug "HACK: Allow to run the index pod as privileged so it has \
-necessary access to run the podman commands."
-  oc -n "$OLM_NAMESPACE" adm policy add-scc-to-user privileged -z default
+    build_image "serverless-bundle" "${rootdir}/olm-catalog/serverless-operator"
 
-  logger.info 'Install the index deployment.'
-  # This image was built using the Dockerfile at 'olm-catalog/serverless-operator/index.Dockerfile'.
-  cat <<EOF | oc apply -n "$OLM_NAMESPACE" -f -
----
-apiVersion: apps/v1
-kind: Deployment
-metadata:
-  name: serverless-index
-  labels:
-    app: serverless-index
-spec:
-  selector:
-    matchLabels:
-      app: serverless-index
-  template:
-    metadata:
-      labels:
-        app: serverless-index
-    spec:
-      containers:
-      - name: registry
-        image: quay.io/openshift-knative/serverless-index:v1.14.3
-        securityContext:
-          privileged: true
-        ports:
-        - containerPort: 50051
-          name: grpc
-          protocol: TCP
-        readinessProbe:
-          exec:
-            command:
-            - grpc_health_probe
-            - -addr=localhost:50051
-        command:
-        - /bin/sh
-        - -c
-        - |-
-          podman login -u $pull_user -p $token image-registry.openshift-image-registry.svc:5000 && \
-          /bin/opm registry add -d index.db --container-tool=podman --mode=replaces -b registry.ci.openshift.org/openshift/openshift-serverless-v$PREVIOUS_VERSION:serverless-stop-bundle,image-registry.openshift-image-registry.svc:5000/$OLM_NAMESPACE/serverless-bundle && \
```

Comment:
I guess removing commands such as `opm registry` also makes us [compatible](https://github.com/redhat-openshift-ecosystem/community-operators-prod/discussions/512) with FBC index.

Reply:
Exactly. The `registry` command is for the old format. We're now using `init` and `render` to generate the new format.